symplyphysics/laws/optics/interference_films_from_angle_refraction_refractive_index_wavelength.py

```diff
@@ -46,7 +47,13 @@ def print_law() -> str:
     refractive_index_=refractive_index,
     angle_refraction_=angle_refraction)
 @validate_output(film_thickness)
-def calculate_film_thickness(wavelength_: Quantity, order_interference_: int, refractive_index_: float, angle_refraction_: float | Quantity) -> Quantity:
+def calculate_film_thickness(wavelength_: Quantity, order_interference_: int | float, refractive_index_: float, angle_refraction_: float | Quantity) -> Quantity:
+    check = order_interference_ * 2 % 2
```

Comment:
On a second thought, it looks like a bad idea. Float division can give quite unpredictable results, so we may face weird issues when trying to distinguish 0.5 from 0.499995.
Let's accept integer number here, but rename it to `double_order_interference` or `number_of_half_waves`. Your newly added checks can be removed, but tests should be updated accordingly to new parameter meaning.